bigchaindb/upsert_validator/validator_election.py

```diff
@@ -141,3 +144,59 @@ def create(cls, tx_signers, recipients, metadata=None, asset=None):
     @classmethod
     def transfer(cls, tx_signers, recipients, metadata=None, asset=None):
         raise NotImplementedError
+
+    @classmethod
+    def to_public_key(cls, election_id):
+        return base58.b58encode(bytes.fromhex(election_id))
+
+    @classmethod
+    def count_votes(cls, election_pk, txns):
+        votes = 0
+        for txn in txns:
```

Comment:
A way to not duplicate the checks is to accept a getter.

```
def count_votes(cls, election_pk, txns, getter=getattr):
    ...
    if getter(txn, 'operation') == 'VALIDATOR_ELECTION_VOTE':
       ...
```
Then you just call `count_votes(election_pk, txns)` for objects and `count_votes(election_pk, txns, operator.itemgetter)` for dictionaries.

On a side note, it would be nice to deal with dictionaries everywhere in the future.